cloudinit/net/netplan.py

```diff
@@ -120,6 +121,18 @@ def _listify(obj, token=" "):
                     "via": subnet.get("gateway"),
                     "to": "default",
                 }
+                subnet_gateway = ipaddress.ip_address(subnet["gateway"])
```

Comment:
I'm being overly cautious, but can we wrap this in a try/except? If we get an exception here (i.e., a bad value passed to us), I don't think it gets caught anywhere, and  I don't think we should crash the entire process over it.

Reply:
OK I wrapped everything around a try/except. `addr` seems to get before though, should I move it outside of the block?

Reply:
Sorry, not sure I follow the question. What do you want to move outside of what block?

Reply:
My fault, I missed a word. I meant that `addr` gets checked before `netplan.py` is called:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cloudinit/cmd/main.py", line 767, in status_wrapper
    ret = functor(name, args)
  File "/usr/lib/python3/dist-packages/cloudinit/cmd/main.py", line 433, in main_init
    init.apply_network_config(bring_up=bring_up_interfaces)
  File "/usr/lib/python3/dist-packages/cloudinit/stages.py", line 940, in apply_network_config
    return self.distro.apply_network_config(
  File "/usr/lib/python3/dist-packages/cloudinit/distros/__init__.py", line 268, in apply_network_config
    network_state = parse_net_config_data(netconfig, renderer=renderer)
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 1103, in parse_net_config_data
    nsi.parse_config(skip_broken=skip_broken)
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 303, in parse_config
    self.parse_config_v1(skip_broken=skip_broken)
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 319, in parse_config_v1
    handler(self, command)
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 115, in decorator
    return func(self, command, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 396, in handle_physical
    subnets = _normalize_subnets(command.get("subnets"))
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 1079, in _normalize_subnets
    return [_normalize_subnet(s) for s in subnets]
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 1079, in <listcomp>
    return [_normalize_subnet(s) for s in subnets]
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 944, in _normalize_subnet
    _normalize_net_keys(
  File "/usr/lib/python3/dist-packages/cloudinit/net/network_state.py", line 994, in _normalize_net_keys
    raise ValueError(f"Address {addr} is not a valid ip address")
ValueError: Address p is not a valid ip address
```

So I thought maybe we could move `subnet_network = ipaddress.ip_network(addr, strict=False)` outside of the `try` block because it should not fail. But we can also leave it for consistency.

Reply:
Gotcha. I think we can keep it as-is for consistency.